Removes or tweaks a shitton of unused, unwanted or simply bad suits and helmets, removes absence of cap's voidsuit

TobyThorne opened the pull request:
Вычищаем мусор из билда, делаем капитана полезным, баффаем оставшуюся броню.

- Переработана броня капитана. Теперь она - войдсьют и имеет свой, новый шлем. Старый шлем остался, но уже больше не космический, но с отличной защитой. Исправлены дырки на месте рук, из-за которых при надевании брони показывались голые руки (ибо перчатки скрываются). Спрайт чутка отредактирован, в особенности - у тонколюдей.
![image](https://user-images.githubusercontent.com/45202681/84257796-bbc4a880-ab37-11ea-924f-ab7c2bf28ac1.png) 
![image](https://user-images.githubusercontent.com/45202681/84258763-48239b00-ab39-11ea-846d-bd8b6d580140.png)
- Переделан спрайт РИГа СБ. Старый - какое-то жуткое месиво, перерисовывать его для тонколюдей не осилил. В итоге собрал заново по кусочкам, адекватно поделил на раздельные части, отрисовал под тонколюдей. От старого почти не отличается.
![image](https://user-images.githubusercontent.com/45202681/84258008-0e9e6000-ab38-11ea-8bd6-8ea0ab50df6d.png)
- Стандартизированы спрайты основных шлемов (обычный, мерковский, абляционный, буллетпруф). Помимо этого, у них теперь есть визор, переключаемый тыком в руке. Функций кроме декоративной не имеет. Для модников.
![image](https://user-images.githubusercontent.com/45202681/84258709-32ae7100-ab39-11ea-9deb-c42954a4d310.png)
- Нахуй выкинуты жилеты со встроенными кармашками. Уже есть модульная система, позволяющая вешать кармашки на обычные жилеты/керриеры.
- Нахуй выкинута тактическая броня. Логики в её присутствии нет - болотный камуфляж в условиях космической станции. С точки зрения геймплея - тоже тупость какая-то, лёгкий бафф обычной брони. 
- Нахуй смержены дублирующиеся жилетики в armor и armor/vest.
- Нахуй выкинуты броньки/наклейки/etc. несуществующих в нынешнем бэке шарашек.
- Нахуй выкинуты тонны неиспользуемых спрайтов из раздела suits, остальное чутка потрогал, но полностью чистить пока что сил не хватит.
- Нахуй выкинуты mirania-шмотки, чем бы они не были.
- Нахуй выкинуты синие варианты СБшной униформы. Это ж не Советы в 2013, ей-богу.
- Пофикшены тонкоспрайты для тяжёлого РИГа Синдиката и коричневого тренча (имелся встроенный галстук, которого нет у обычного спрайта).
- Оба варианта плаща детектива переименованы в "detective's trenchcoat", обновлено описание.
- Изменены лодаутные плащи. Используют старые названия, новое описание и ничтожные показатели брони. Нынешние 50 мили-защиты из лодаута - как-то не очень.
- Куски кода брони более-менее отсортированы в соответствующие файлы из разных зажопных местечек.
- Содержимое ящиков СБ, детектива, вардена, ХоСа, ХоПа и капитана отсортировано по типам.
- Обычные серые скафандры больше не замедляют.
- Аварийные скафандры замедляют на 3, а не на 4.

Попутно откалиброваны показатели брони в сторону стандартизации. И полезности специализированной брони из оружейки.
Формат: MELEE, BULLET, LASER.
- Обычная броня (и средняя плитка): 40, 40, 40
- Тяжёлая плитка: 60, 60, 60
- Плащ детектива: 35, 35, 35 (защищает плечи)
- Плащ вардена: 50, 50, 50 (защищает плечи и колени)
- Шмот ХоСа: 65, 65, 65
- Скафандр СБ: 60, 40, 40
- Скафандр ХоСа: 70, 50, 50
- Скафандр капитана: 75, 60, 60
- РИОТ-броня: 80, 35, 35
- Баллистическая броня: 35, 85, 35
- Абляционная броня: 35, 35, 85
Последние три наконец стали полезными, ибо дают почти полную защиту от соответствующего типа урона. Ну и стандартную - от остальных.

- Попутно добавлены недостающие лампочки и выключатели на карту.
- Из под-оружейной убраны тяжёлые плитки по просьбе геймдизов.
- Из стрельбища убрана одна коробка с дымовыми.
- В под-оружейную добавлены две коробки с дымовыми.

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Commit:
armor cleanup

- 2much4desc

Commit:
Update armor.dm

kokokobold commented:
>Нахуй выкинуты синие варианты СБшной униформы. Это ж не Советы в 2013, ей-богу.

Ужасно.

MrMagick2104 commented:
> Плащ детектива: 40, 40, 40

Не жирно давать детективу бронежилет ++ вместо плаща? Лучше бы как обычный бронежилет его сделать, имхо. Детективу он нужен для защиты, а не как улучшенная броня СБ, нет?

Commit:
minor tweaks

- Soft suits, unlike voidsuits, no longer slow the wearer down
- Emergency suit slowdown reduced from 4 to 3
- Det's trench overall armor lowered by 15

TobyThorne commented:
- Дефективское пальто снижено до уровня обычного жилета (35, 35, 35)
- У обычных скафандров убрано замедление, давая им преимущество перед войдсьютами.
- Замедление аварийных скафандров снижено с 4 до 3. Учитывая замедления от всяких ранок и прочего, использовать их практически невозможно.

MrMagick2104 commented:
> У обычных скафандров убрано замедление, давая им преимущество перед войдсьютами.

Обычные скафандры - это кто?
Софтсьюты?

PlinHost commented:
Линк на иссуй автоматически закроет иссуй при принятии ПРа, спасибо, гитхаб.

TapKo4eB commented:
напоминаю
![изображение](https://user-images.githubusercontent.com/48761132/84769337-57459580-afde-11ea-991c-b1ee8a5e475e.png)

PlinHost commented:
virgin броня детектива:
UPPER_TORSO|LOWER_TORSO
melee = 35, bullet = 35, laser = 35, energy = 15, bomb = 25, bio = 0, rad = 0

chad пальто детектива:
slots = 2, slot_size = 2
UPPER_TORSO|LOWER_TORSO|ARMS
melee = 35, bullet = 35, laser = 35, energy = 20, bomb = 25, bio = 0, rad = 0

Commit:
Merge branch 'dev' of https://github.com/ChaoticOnyx/OnyxBay into armorcleanup

Commit:
Update

TobyThorne commented:
ПР обновлён.
Броня жилетика поднята до 40/40/40, детективский плащ остался с 35/35/35, добавил лампочек и выключателей туда, где их не хватало.

Commit:
Возвращение блудных кастомок

- Случайно выпеленные кастомки возвращены и поселены в отдельный файл.
Ради всего святого, подписывайте своё говно, хер знает как узнавать, что это - кастомки.

Commit:
Update baystation12.dme

kokokobold commented:
>Нахуй выкинуты синие варианты СБшной униформы. Это ж не Советы в 2013, ей-богу.

Категорически против. Цвет хороший, зачем выпиливать?

>/obj/item/clothing/suit/storage/toggle/forensics/customred //Custom item

Тоже кастомка Булата вообще-то.
А так изменения хорошие во многом, одобряю!

TobyThorne commented:
> Категорически против. Цвет хороший, зачем выпиливать?

Потому что не используется, а только захламляет билд. Если вдруг кто-то захочет полностью перекрасить СБ в синий - нарисовать/найти униформу будет не самой сложной проблемой.

> Тоже кастомка Булата вообще-то.

Да блядь. Когда уже подписывать научатся.

Commit:
Кастомка булата

- Попутно отрисовал спрайты на ебучие бодитайпы и пофиксил исчезновение спрайта при расстёгивании. Сука качество.

Commit:
Окошкофикс

- Добавлен пол под окна в техах у отбытия.

TobyThorne commented:
- Добавлен пол под окна в техах у отбытия.

TapKo4eB commented on code/modules/clothing/under/jobs/security.dm:
```diff
@@ -128,111 +129,28 @@
+/obj/item/clothing/under/rank/head_of_security/jensen
+	desc = "You never asked for anything that stylish."
+	name = "head of security's jumpsuit"
+	icon_state = "jensen"
```
`item_state = "jensen"` забыл
![изображение](https://user-images.githubusercontent.com/48761132/88058145-bcefe900-cb6b-11ea-8026-2b155e215be5.png)

TobyThorne replied:
Добавил item_state_slots = list(slot_hand_str = "jensen")

TapKo4eB commented on code/modules/mob/living/bot/ed209bot.dm:
```diff
@@ -103,7 +103,7 @@
 
 		if(2)
-			if(istype(W, /obj/item/clothing/suit/storage/vest) || istype(W, /obj/item/clothing/suit/armor/pcarrier) || istype(W, /obj/item/clothing/accessory/armorplate) || istype(W, /obj/item/clothing/suit/armor/vest/old))
+			if(istype(W, /obj/item/clothing/suit/storage/vest) || istype(W, /obj/item/clothing/suit/armor/pcarrier) || istype(W, /obj/item/clothing/accessory/armorplate))
```
удалил весты но почему-то кучу связанного кода с ним (`/obj/item/clothing/suit/storage/vest`) оставил.
/obj/item/clothing/suit/armor/vest/old на /obj/item/clothing/suit/armor/vest мб

TobyThorne replied:
Сторажи почистил, vest/old вроде и так нигде не остался.

TobyThorne commented on code/modules/clothing/custom_clothing.dm:
```diff
@@ -0,0 +1,62 @@
+	item_state = "dalek_coat"
+	armor = list(melee = 45, bullet = 35, laser = 35, energy = 10, bomb = 10, bio = 0, rad = 0)
+
+/obj/item/clothing/suit/armor/hos/jensen/fieldcoat/mob_can_equip(mob/user)
+	.=..()
+	if(user.gender == FEMALE)
+		to_chat(user, SPAN("warning", "You aren't sure you'll fit in this fascist cloth..."))
+		return 0
```
Это чужой кастомный чудокод, не хочу его трогать.

TobyThorne replied:
Скорее всего, проверяет на каждый слот.

TobyThorne commented on code/modules/clothing/head/helmet.dm:
```diff
@@ -10,124 +10,61 @@
+	update_clothing_icon()
 
-/obj/item/clothing/head/helmet/nt/guard
+/obj/item/clothing/head/helmet/nt
```
Оно давным-давно в таком состоянии, всем норм вроде. Другое дело - зачем вообще нужны дебильные нашлемные тряпки.

TobyThorne commented on code/modules/clothing/suits/armor.dm:
```diff
@@ -459,9 +281,9 @@
-	item_state = "m_vest"
-	armor = list(melee = 5, bullet = 30, laser = 20, energy = 20, bomb = 25, bio = 0, rad = 0)
\ No newline at end of file
+/obj/item/clothing/suit/armor/hos/jensen/custom
```
Не был уверен, что это именно опухшая кастомка.

TobyThorne replied:
Переместил.

Commit:
Fixes everything

- Fixed SWAT coat turning a spessman into an exhibitionist
- Civilian trench is lo longer a subtype of det trench, and thus cannot hold weapons/ammo in the suit storage slot
- Cleaned up the remaining suit/storage/vest cases
- Moved some custom items into custom_clothing.dm
- Returned police helmet (custom item) sprite
- Fixed test subjects' headgear and pirate helmets having has_visor = 1

TobyThorne commented on code/modules/clothing/custom_clothing.dm:
```diff
@@ -0,0 +1,62 @@
+/obj/item/clothing/head/helmet/police
+	name = "police helmet"
+	desc = "It's a helmet specifically designed for police use. Comfortable and robust."
+	icon_state = "helmet_police"
```
Поправил.

TobyThorne commented on code/modules/clothing/suits/armor.dm:
```diff
@@ -20,147 +19,53 @@
+	item_state = "hostrench"
+	flags_inv = 0
+
+/obj/item/clothing/suit/armor/hos/german
```
Поправил.

TobyThorne commented on code/modules/clothing/suits/armor.dm:
```diff
@@ -408,24 +162,92 @@
-	armor = list(melee = 50, bullet = 30, laser = 20, energy = 10, bomb = 25, bio = 0, rad = 0)
+	armor = list(melee = 90, bullet = 90, laser = 90, energy = 55, bomb = 90, bio = 50, rad = 100)
+
+/obj/item/clothing/suit/armor/swat/officer
```
Исправил.

Commit:
quickfix

- Fixed ED209 crafting recipe;
- Removed 1 box of smoke grenades from the shooting range;
- Added 2 boxes of smoke grenades to the sub-armory;
- Removed merc armor plates from the sub-armory;

TapKo4eB reviewed (dismissed):
ну вроде приколов больше по коду не нашёл

Ruforce commented:
В чём прикол желания выпиливать ВСЁ? 
Пиздец какой-то, лучше бы баги шёл фиксить

Ruforce commented:
Кастомный жилет татарина не вижу. Который vest/police.

Шлем есть, плитника нет.

TapKo4eB commented on code/modules/clothing/suits/armor.dm:
```diff
@@ -190,157 +82,6 @@
-	max_heat_protection_temperature = ARMOR_MAX_HEAT_PROTECTION_TEMPERATURE
-	siemens_coefficient = 0.6
-
-/obj/item/clothing/suit/storage/vest/police
-	name = " police armored vest"
-	desc = "A synthetic armor vest with a large webbing and additional ballistic plates. Has a name badge on the frontal plate, that reads 'Sgt. Bauer'"
-	icon_state = "policevest"
-	item_state = "policevest"
-	armor = list(melee = 40, bullet = 40, laser = 45, energy = 15, bomb = 30, bio = 0, rad = 0)
```
кастомки в кустах

Commit:
police vest

TapKo4eB commented:
подожди-ка секунду

>Нахуй выкинуты жилеты со встроенными кармашками. Уже есть модульная система, позволяющая вешать кармашки на **обычные жилеты**/керриеры.

мб ты имеешь под жилетами ввиду джапсьюты а может быть ты немного ошибаешься т.к.
```
/obj/item/clothing/suit/armor/vest
	name = "armor"
	desc = "An armored vest that protects against some damage."
	icon_state = "onyxvest"
	(...)
	armor = list(melee = 40, bullet = 40, laser = 40, energy = 15, bomb = 25, bio = 0, rad = 0)
	valid_accessory_slots = list(ACCESSORY_SLOT_INSIGNIA) <<<========
```

Ruforce commented:
И карманы кастомке верни. Что за мода пошла - выпиливать все карманы?

TapKo4eB commented:
их там две штуки

TapKo4eB reviewed (approved):
ладно потом докачаем допилы
